test/interactions/clickInteractionTests.ts

```diff
@@ -1,158 +1,208 @@
 ///<reference path="../testReference.ts" />
 
-describe("Interactions", () => {
-  describe("Click Interaction", () => {
-    describe("Basic Usage", () => {
-      let SVG_WIDTH = 400;
-      let SVG_HEIGHT = 400;
+describe("Click Interaction", () => {
+  const SVG_WIDTH = 400;
+  const SVG_HEIGHT = 400;
 
-      let svg: d3.Selection<void>;
-      let component: Plottable.Component;
-      let clickInteraction: Plottable.Interactions.Click;
+  let svg: d3.Selection<void>;
+  let component: Plottable.Component;
+  let clickInteraction: Plottable.Interactions.Click;
 
-      beforeEach(() => {
-        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-        component = new Plottable.Component();
-        component.renderTo(svg);
+  beforeEach(() => {
+    svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+    component = new Plottable.Component();
+    component.renderTo(svg);
 
-        clickInteraction = new Plottable.Interactions.Click();
-        clickInteraction.attachTo(component);
-      });
+    clickInteraction = new Plottable.Interactions.Click();
+    clickInteraction.attachTo(component);
+  });
```

Comment:
One thing that we have seen through our test refactors is that having a `beforeEach` at the top of the suite can be a bit annoying.  One is that if an inner describe also declares a `beforeEach`, it becomes harder to track the code flow.  Also, if the `beforeEach` is invalid for some tests, then it becomes a major hassle to work around the `beforeEach()`.  What are your thoughts on this?